compiler/rustc_mir/src/borrow_check/consumers.rs

```diff
@@ -0,0 +1,37 @@
+//! This file provides API for compiler consumers.
```

Comment:
how stable do you hope this API to be ? whether it'd be the polonius facts structure (e.g. relation names, or types), their contents, or the polonius output

Reply:
For Prusti, we are upgrading the Rust compiler version twice a month. My hope is that we do not spend more than 3-4 person-hours per upgrade. (Currently, it is about 1 hour on average.)

Reply:
By the way, Prusti is already using Polonius, but in a very hacky way. So, this PR should only reduce our maintenance efforts.

Reply:
Ok.

I'll add this as a point of discussion for next week's polonius sprint (I didn't expect we'd have users that rely on all these polonius implementation details) and probably need to maintain some stability to not break prusti.

Reply:
> I'll add this as a point of discussion for next week's polonius sprint (I didn't expect we'd have users that rely on all these polonius implementation details) and probably need to maintain some stability to not break prusti.

Thanks! However, I would not worry too much because I think that the data we need is fundamental (in the sense that you will not be able to implement the borrow checker without it).

By the way, feel free to tag me on the discussion. I will be happy to comment on whether the planned changes affect us.

Reply:
I mean more for hypotheticals like changing implementation like we talked about in barcelona: "what if we need to change the whole facts to something other than flat relations", or "don't compute all the loans in all the origins at all points", or "what if we move away from datalog one day" that kinda thing 😅 

In the close future I wouldn't worry too much about that, for sure.

Reply:
In the long term, the goal would be to make Prusti use `optimized_mir` or some other version of MIR that is used by all compiler consumers. Then how Polonius works should not matter anymore.